### ISSUE TITLE
Vsimem parse error when reading shapefiles from minio s3 server

### ISSUE THREAD

**Issue by forestbat** [OSGeo/gdal#9042]
https://github.com/geopandas/geopandas/issues/3129

I try to read my shapefile from my minio s3 server，however it crashed with this：
```
# fiona.errors.DriverError: '/vsimem/965167b805054f868869be46983490b5' not recognized as a supported file format.
# pyogrio.errors.DataSourceError: '/vsimem/9ff509ef6970482e959c0a468b46df87' not recognized as a supported file format.
```
Because of `vsimem` is head of GDAL's virtual file system, so I think this is because of GDAL's problem on minio s3 server.

Please fix this problem or call me where I can solve it.

## Operating system

Windows Server 2022

## GDAL version and provenance

3.6.2 from Windows Server 2022 Datacenter 21H2

**Comment by jratike80** [comment]
What was the result of these tests which were recommended:

> What happens of you put a zip containing all files for the shapefile on your minio instance and try to read that instead? Or use one of the single-file formats, like GPKG, or FlatGeobuf?

**Comment by jratike80** [comment]
I installed Minio on my Windows laptop, created a bucket and uploaded some shapefiles into the bucket. Ogrinfo can read the shapefiles without any troubles. The command that I used:

`ogrinfo /vsis3/geodata/GB_Postcodes/PostalArea.shp --config AWS_NO_SIGN_REQUEST YES --config AWS_S3_ENDPOINT 192.168.1.22:9000 --config AWS_HTTPS NO --config AWS_VIRTUAL_HOSTING FALSE -al`

My GDAL version is GDAL 3.9.0dev.
